lib/commit-lint.ts

```diff
@@ -0,0 +1,152 @@
+import { commentFunction } from "./ci-helper";
+import { IPRCommit } from "./github-glue";
+
+/*
+ * Simple class to drive lint tests on commit messages.
+ */
+export class LintCommit {
+    private readonly addComment: commentFunction;
+
+    public constructor(addComment: commentFunction) {
+        this.addComment = addComment;
+    }
+
+    /*
+     * Linter method to run checks on the commit message.
+     * @param {IPRCommit} the patch to be checked
+     *
+     * The lintings are in methods called from here.  If
+     * a lint check is too severe to continue, it will throw
+     * an error.
+     */
+    public async lint(patch: IPRCommit):
+        Promise<boolean> {
+        let result: boolean = true;
+        const lines =  patch.message.split("\n");
+        const msgs: string[] = [];
+
+        try {
+           // add a line for each test
+
+            // tslint:disable:curly
+           if (!await this.lint1(lines, msgs)) result = false;
+           if (!await this.lint2(lines, msgs)) result = false;
+           if (!await this.lint3(lines, msgs)) result = false;
+           if (!await this.lint4(lines, msgs)) result = false;
+
+            // tslint:enable:curly
+        } catch (error) {
+            msgs.unshift(`Commit message checking stopped due to:\n${error}`);
+            result = false;
+        }
+
+        if (msgs.length) {
+            await this.addComment(`There ${
+                    msgs.length > 1 ?
+                    "are issues" : "is an issue"
+                    } in commit ${
+                patch.commit}:\n${msgs.join("\n")}`);
+        }
+
+        return result;
+    }
+/*
+TODO:
+There is also the convention that the commit message should talk in the
+imperative tone (as opposed to the past tense, as many other projects prefer).
+I am not sure that there is an easy way to test for that.
```

Comment:
I think we can (and should) leave this to a future commit. Either we can use the Stanford Parser, as the `node-commit-msg` package does (which would require Java, but that's fine, I think), or we can go for something much, much simpler, e.g. imitating how [ELIZA](https://en.wikipedia.org/wiki/ELIZA) parses the English language to detect whether the sentence uses the present or past tense.

Reply:
Absolutely.  I had copied the initial list from the issue and moved the items as they were implemented.  I am removing the comment as there will still be an issue open for it.